realm/realm-library/src/objectServer/java/io/realm/RealmApp.java

```diff
@@ -98,14 +85,95 @@ public RealmApp(String appId) {
      * @param config
      */
     public RealmApp(RealmAppConfiguration config) {
+        // FIXME: Right now we only support one RealmApp. This class will throw a
+        // exception if you try to create it twice. This is a really hacky way to do this
+        // Figure out a better API that is always forward compatible
+        synchronized (SyncManager.class) {
+            String appId = config.getAppId();
+            if (CREATED_APP_ID != null && !appId.equals(CREATED_APP_ID)) {
+                throw new IllegalStateException("Only one RealmApp is currently supported. " + CREATED_APP_ID + " already exists.");
+            }
+            CREATED_APP_ID = appId;
+        }
+
```

Comment:
Yes. I guess we should somehow manage them in Realm, but don't what differentiates multiple apps when we want to support more than one. Is it only _app id_? And are they never a need to clearing them?

Reply:
And is it safe to have multiple instance? Seems like we can create multiple instances as long as it just have the same _app id_.

Reply:
I modified the check, so only one instance is created. Also improved the error message.